src/main/java/io/cdap/e2e/pages/actions/CdfConnectionActions.java

```diff
@@ -0,0 +1,204 @@
+    int attempts = 0;
+    while (attempts < 5) {
+      try {
+        ElementHelper.hoverOverElement(CdfPluginPropertiesLocators.locateElementWithText(dataRow));
+        ElementHelper.clickOnElement(CdfPluginPropertiesLocators.locateElementWithText(dataRow));
+        break;
+      } catch (StaleElementReferenceException e) {
+        if (attempts == 4) {
+          throw e;
+        }
+      }
+      attempts++;
+    }
+  }
+
+  /**
+   * Select data for connection using name
+   *
+   * @param dataName If dataName is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as a key
+   *                 then actual dataName to select is fetched from it
+   *                 else dataName used as it is.
+   */
+  public static void selectConnectionDataWithName(String dataName) {
+    String dataToSelect = PluginPropertyUtils.pluginProp(dataName);
+    if (dataToSelect == null) {
+      dataToSelect = dataName;
+    }
+    selectConnectionDataRow(dataToSelect);
+  }
+
+  /**
+   * Select data rows for connection traversing through directories using provided path
+   *
+   * @param dataPath If dataPath is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as a key
+   *                 then actual dataPath to select is fetched from it
+   *                 else dataPath used as it is.
+   */
+  public static void selectConnectionDataWithPath(String dataPath) {
+    String dataPathToSelect = PluginPropertyUtils.pluginProp(dataPath);
+    if (dataPathToSelect == null) {
+      dataPathToSelect = dataPath;
+    }
+    String[] dataPathSplit = dataPathToSelect.split("/");
+    for (String dataRow : dataPathSplit) {
+      selectConnectionDataRow(dataRow);
+    }
+  }
+
+  /**
+   * Verify the connection input datatable is displayed with title
+   *
+   * @param dataPath If dataPath is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as a key
+   *                 then actual dataPath is fetched from it
+   *                 else dataPath used as it is.
+   */
+  public static void verifyTheConnectionDataTableDisplayed(String dataPath) {
+    String actualDataPath = PluginPropertyUtils.pluginProp(dataPath);
+    if (actualDataPath == null) {
+      actualDataPath = dataPath;
+    }
+    String[] dataPathSplit = actualDataPath.split("/");
+    String title = dataPathSplit[dataPathSplit.length - 1];
+    WaitHelper.waitForElementToBePresent(CdfConnectionLocators.locatorOfDataTableTitle(title));
+    WaitHelper.waitForElementToBeDisplayed(CdfConnectionLocators.dataTable);
+  }
+
+  /**
+   * Click on Create Pipeline button in Wrangler datatable page
+   */
+  public static void clickCreatePipelineButton() {
+    ElementHelper.clickOnElement(CdfConnectionLocators.createPipelineButton);
+  }
+
+  /**
+   * Select the type of pipeline to create
+   *
+   * @param pipelineType If pipelineType is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as key
```

Comment:
I think priority should be `inverse`, if the property is passed as a param select it otherwise from `Properties File`.

Reply:
Priority is correct.
As we are keeping actual values in the properties file. 

Example - 
`Then Enter input plugin property: "datasetProjectId" with value: "projectId"`

In this case if we `inverse` priority. It will enter value as "projectId" in input instead of actual value "cdf-athena" from properties file.

Reply:
I see, we are passing plugin_prop as the param.